Newspack/Newspack/Controllers/AssetsViewController.swift

```diff
@@ -1,39 +1,129 @@
 import UIKit
+import CoreData
 import WordPressFlux
 
 class AssetsViewController: UITableViewController {
 
-    var items = [URL]()
-    var receipt: Any?
+    var dataSource: AssetDataSource!
 
     override func viewDidLoad() {
         super.viewDidLoad()
 
-        items = StoreContainer.shared.folderStore.listCurrentStoryFolderContents()
-
-        receipt = StoreContainer.shared.folderStore.onChange { [weak self] in
-            self?.items = StoreContainer.shared.folderStore.listCurrentStoryFolderContents()
-            self?.tableView.reloadData()
-        }
+        configureDataSource()
     }
 
-    // MARK: - Table view data source
+}
+
+// MARK: - Actions
+
+extension AssetsViewController {
 
-    override func numberOfSections(in tableView: UITableView) -> Int {
-        return 1
+    @IBAction func handleAddTapped(sender: Any) {
+        let action = AssetAction.createAssetFor(text: "New Text Note")
+        SessionManager.shared.sessionDispatcher.dispatch(action)
     }
 
-    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return items.count
+}
+
+// MARK: - TableViewDelegate methods
+extension AssetsViewController {
+
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+
     }
+}
 
-    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: "AssetCell", for: indexPath)
+// MARK: - DataSource related methods
+extension AssetsViewController {
 
-        let url = items[indexPath.row]
-        cell.textLabel?.text = url.lastPathComponent
+    func cellFor(tableView: UITableView, indexPath: IndexPath, storyAsset: StoryAsset) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "AssetCell", for: indexPath)
```

Comment:
**No need to change anything!!!** I love [this trick](https://github.com/Automattic/simplenote-macos/blob/develop/Simplenote/NSTableViewCell+Simplenote.swift#L11)!!!

Reply:
I [like that trick too](https://github.com/Automattic/newspack-ios/blob/develop/Newspack/Newspack/Extensions/UITableViewCell%2BHelpers.swift). :D  For now this is just a regular UITableViewCell vs a custom class. When I create the custom cells I'll up date this too.   Also props to you for originally suggesting the trick several PRs ago!